src/util/Recoil_deepFreezeValue.js

```diff
@@ -56,6 +56,10 @@ function shouldNotBeFrozen(value: mixed): boolean {
     return true;
   }
 
+  if (typeof window !== 'undefined' && value instanceof window) {
```

Comment:
This should check for instance of `Window` vs `window`.  I can fix that and also normalize the `typeof` check.

Reply:
Ok, I have a fix for the internal diff; though note it won't back-sync to this PR.